smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -227,6 +227,7 @@ private void generateOperationSerializer(
                 HttpProtocolGeneratorUtils.writeHostPrefix(context, operation);
             }
 
+            writer.write("const endpoint = {...context.endpoint, path: undefined, query: undefined};");
```

Comment:
Can you add comments to these lines explaining why the indirection is necessary?